crates/holochain/src/core/workflow/countersigning_workflow/incomplete.rs

```diff
@@ -0,0 +1,336 @@
+                                        if cs.preflight_request == session_data.preflight_request {
+                                            tracing::debug!("Agent {:?} has a countersigning entry for the same session", agent);
+                                            // This agent appears to have completed the countersigning session.
+                                            // Collect the signed action to use as a signature for completing the session for our agent.
+                                            SessionCompletionDecision::Complete(Box::new(
+                                                maybe_countersigning_record.signed_action.clone(),
+                                            ))
+                                        } else {
+                                            // This is evidence that the other agent has put something else on their chain.
+                                            // Specifically, some other countersigning session.
+                                            SessionCompletionDecision::Abandoned
+                                        }
+                                    }
+                                    RecordEntry::Present(_) => {
+                                        // Anything else on the chain is evidence that the agent did not complete the countersigning.
+                                        tracing::debug!(
+                                            "Agent {:?} has a non-countersigning entry",
+                                            agent
+                                        );
+                                        SessionCompletionDecision::Abandoned
+                                    }
+                                    RecordEntry::Hidden | RecordEntry::NA => {
+                                        // This wouldn't be the case for a countersigning entry so this is evidence that
+                                        // the agent has put something else on their chain.
+                                        SessionCompletionDecision::Abandoned
+                                    }
+                                    RecordEntry::NotStored => {
+                                        // This case is not determinate. The agent activity authority might
+                                        // have the action but not the entry yet. We can't make a decision here.
+                                        // In this case, we will also have tried to ask a record authority for
+                                        // this missing entry.
+                                        SessionCompletionDecision::Indeterminate
+                                    }
+                                }
+                            }
+                        }
+                        _ => {
+                            tracing::warn!("Agent authority returned an unexpected response for agent {:?}: {:?}", agent, activity);
```

Comment:
Is this what we'll see in the case where the AAA only returned the action and for whatever reason we couldn't fetch the entry that goes with it?

Reply:
Oh, nevermind. I see that in the above arm `NotStored`.

Reply:
This would be if we were talking to either a misbehaving conductor or an older conductor that was returning a `::Hashes` response instead of `Full`. Or even a newer conductor that had some other behavior again. We can't really do anything useful without the entry so I've just left this as "can't use the response"